modules/tensor_mechanics/src/materials/ComputeThermalExpansionEigenStrain.C

```diff
@@ -24,6 +24,7 @@ ComputeThermalExpansionEigenStrain::ComputeThermalExpansionEigenStrain(const Inp
     _has_incremental_strain(hasMaterialProperty<RankTwoTensor>(_base_name + "strain_increment")),
     _temperature_old(_has_incremental_strain ? & coupledValueOld("temperature") : NULL),
     _thermal_expansion_coeff(getParam<Real>("thermal_expansion_coeff")),
+    _dstrain_dT(declareProperty<RankTwoTensor>(_base_name + "dstrain_dT")),
```

Comment:
This class couples to temperature and should use `declareMaterialPropertyDerivative`